### PR TITLE
[#9536] Feedback Text Question E2E Test

### PR DESCRIPTION
**Pull request by jtankw3**
Part of #9536

**Outline of Solution**

- Add backdoor method to get FeedbackResponse
- Edit page functions:
    - verify loaded question
    - add new question
    - copy question
    - edit question
- Submit page functions:
    - verify loaded question
    - submit response
    - check previous response
    - edit response

### CODE REVIEW

File: src/e2e/java/teammates/e2e/cases/e2e/FeedbackTextQuestionE2ETest.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+
+        verifyPresentInDatastore(response);
+
+        ______TS("check previous response");
+        feedbackSubmitPage = AppPage.getNewPageInstance(browser, url, FeedbackSubmitPage.class);
+        feedbackSubmitPage.waitForPageToLoad();
+        feedbackSubmitPage.verifyTextResponse(1, receiver.getName(), response);
+
+        ______TS("edit response");
+        String editedResponse = "<p><strong>Edited response</strong></p>";
```
Comment (rrtheonlyone) [inline src/e2e/java/teammates/e2e/cases/e2e/FeedbackTextQuestionE2ETest.java:117]: Minor suggestion: Shall we make this string longer to see if the response length field can work? Maybe a string that has 100 words?

**Review by rrtheonlyone — CHANGES_REQUESTED** [review]
Great work with this!

Some additional test cases that we perhaps can look into:
- setting recommended length to 0 or negative number (do we want to test those?)
- changing evaluee/recipient dropdown when submitting feedback 

Also, I realised we have not touched on feedback path/visibility. Although, I think this is not the best place to test it (since they are the same for all questions).

**Comment by jtankw3** [comment]
> Great work with this!
> 
> Some additional test cases that we perhaps can look into:
> 
> * setting recommended length to 0 or negative number (do we want to test those?)
> * changing evaluee/recipient dropdown when submitting feedback
> 
> Also, I realised we have not touched on feedback path/visibility. Although, I think this is not the best place to test it (since they are the same for all questions).

Hi @rrtheonlyone, thank you very much for your review.

Firstly, as this is E2E testing, I think we may want to avoid doing negative testing as we are more concerned about testing for integration and data integrity. I believe the validation test for the different fields should probably be done in a separate, less expensive unit test.

As for feedback path/visibility, I have tested question visibility and the recipient list in the more general `FeedbackSubmitPageE2ETest`. I have also tried to vary the receiver type across the 10 tests for different question types so that there is larger test coverage. 

In particular for this test, receiver type is instructor with a limited `numberOfEntitiesToGiveFeedbackTo`. As there is a limited `numberOfEntitiesToGiveFeedbackTo`, the dropbox is used when selecting the receiver for the response so I think that is sufficient for testing. Or do you think we should test changing the recipient after submission and submitting again? Not sure if that would make a difference.

File: src/e2e/java/teammates/e2e/cases/e2e/FeedbackTextQuestionE2ETest.java (hunk 1 above)
Comment (jtankw3) [reply src/e2e/java/teammates/e2e/cases/e2e/FeedbackTextQuestionE2ETest.java:117]: Actually above there is a string of length 7 that aims to test the counting functionality of the response length field. I believe increasing it to 100 should not make a difference unless we are testing for handling of excess number of words,  which I think should not be done for E2E test.

**Review by rrtheonlyone — APPROVED** [review]
Yes I see where you are coming from! I think that is reasonable :) 

I think this PR is good then 👍

**Review by wkurniawan07 — APPROVED** [review]
Actually, negative testing should also be part of E2E testing, because negative user input is still part of the normal user flow (in a way). What is true, however, is that you do not need to try more than one negative test; just do one set of negative input, check that error is handled as expected, and proceed. No need to handle different sets of negative inputs; those are the job of lower level tests, as you have mentioned.

Having said the above, this PR can proceed as is (good job!) and we can add the negative tests in later date.
